File: test/new_lang/test3/test_program.cpp
#include "splashkit.h"

#include <iostream>
using namespace std;

int main()
{
  cout << get_int() << endl;
  cout << get_int(1) << endl;
  cout << get_uint(2) << endl;
  cout << get_short(3) << endl;
  cout << get_ushort(4) << endl;
  cout << get_float(5.55) << endl;
  cout << get_double(6.66) << endl;
  cout << get_long(7) << endl;
  cout << get_char('A') << endl;
  cout << get_uchar('a') << endl;
  cout << get_bool(true) << endl;
  cout << get_bool(false) << endl;
}

File: test/new_lang/test3/test_program.rs
#[path = "../../out/rust/splashkit.rs"]
mod splashkit;
use splashkit::*;

fn main() {
  println!("{}", get_int());
  println!("{}", get_int_with_initial_value(1));
  println!("{}", get_uint(2));
  println!("{}", get_short(3));
  println!("{}", get_ushort(4));
  println!("{}", get_float(5.55));
  println!("{}", get_double(6.66));
  println!("{}", get_long(7));
  println!("{}", get_char('A'));
  println!("{}", get_uchar('a'));
  println!("{}", get_bool(true));
  println!("{}", get_bool(false));
}
